### PR TITLE
CEDS-1476: Remove old caching from controllers and tests.

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**

### CODE REVIEW

File: app/services/cache/ExportsCacheModelRepository.scala [hunk 1]
```diff
@@ -69,6 +70,8 @@ class ExportsCacheModelRepository @Inject()(mc: ReactiveMongoComponent, appConfi
         updateResult.result[ExportsCacheModel]
       }
 
+  def remove(sessionId: String): Unit = collection.findAndRemove(selector = bySessionId(sessionId)).map(_ => ())
```
Comment (bambuchaAdm) [inline app/services/cache/ExportsCacheModelRepository.scala]: Why we ignoring result future with this - even logging of errors ?

File: app/services/SubmissionService.scala [hunk 2]
```diff
@@ -55,12 +56,11 @@ class SubmissionService @Inject()(
     auditService.auditAllPagesUserInput(getCachedData(exportsCacheModel))
     exportsConnector.submitExportDeclaration(data.ducr, data.lrn, data.payload).flatMap {
       case HttpResponse(ACCEPTED, _, _, _) =>
-        cacheService.remove(cacheId).map { _ =>
-          auditService.audit(AuditTypes.Submission, auditData(data.lrn, data.ducr, Success.toString))
-          exportsMetrics.incrementCounter(submissionMetric)
-          timerContext.stop()
-          data.lrn
-        }
+        cacheService.remove(sessionId)
```
Comment (bambuchaAdm) [inline app/services/SubmissionService.scala]: Here we have little change of semantic - what if in removing from cache there goes something wrong - we don't have even know abut it.

File: app/models/declaration/governmentagencygoodsitem/GovernmentAgencyGoodsItem.scala [hunk 3]
```diff
@@ -90,23 +87,4 @@ case class GovernmentAgencyGoodsItemAdditionalDocumentSubmitter(
...
-case class DateTimeString(formatCode: String, value: String) {
-  import DateTimeFormats._
-
-  def time(): ZonedDateTime = formatCode match {
-    case "102" => handle102Format(value)
-    case "304" => handle304Format(value)
-  }
-}
-
-object DateTimeFormats {
```
Comment (bambuchaAdm) [inline app/models/declaration/governmentagencygoodsitem/GovernmentAgencyGoodsItem.scala]: Why this glue-code goes out?

Comment (mattclark-zerogravit) [reply app/models/declaration/governmentagencygoodsitem/GovernmentAgencyGoodsItem.scala]: I checked for usages and couldn't find any

Comment (PatrykRudnicki) [reply app/models/declaration/governmentagencygoodsitem/GovernmentAgencyGoodsItem.scala]: Those formats was related with wco-dec. Hopefully now we're handling date in different (correct) way. Those two formats are formats that you can find in your declaration. I don't remember why we handled this in way above but I'm sure it was related with wco-dec

Comment (mattclark-zerogravit) [reply app/models/declaration/governmentagencygoodsitem/GovernmentAgencyGoodsItem.scala]: We persisted wco models in the cache.. so I think these formats were used to convert the dates before they were cached.

File: app/services/SubmissionService.scala (hunk 2 above)
Comment (mattclark-zerogravit) [reply app/services/SubmissionService.scala]: Have changed how we do this so all code in controller is handled inside the future.

File: app/services/cache/ExportsCacheModelRepository.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply app/services/cache/ExportsCacheModelRepository.scala]: Have changed how we do this so all code in controller is handled inside the future.

**Pull request merged**